source/slang/core.meta.slang

```diff
@@ -1347,12 +1347,12 @@ struct TextureTypeInfo
             
         if(prefixInfo.combined)
         {
-            sb << "__magic_type(TextureSampler," << int(flavor) << ")\n";
+            sb << "__magic_type(TextureSamplerType," << int(flavor) << ")\n";
             sb << "__intrinsic_type(" << (kIROp_TextureSamplerType + (int(flavor) << kIROpMeta_OtherShift)) << ")\n";
         }
         else
         {
-            sb << "__magic_type(Texture," << int(flavor) << ")\n";
+            sb << "__magic_type(TextureType," << int(flavor) << ")\n";
             sb << "__intrinsic_type(" << (kIROp_TextureType + (int(flavor) << kIROpMeta_OtherShift)) << ")\n";
```

Comment:
Similarly, having the "flavor" bits here for texture types is increasingly seeming like an unnecessary complication.

Eventually I think we will want to shift to having a single generic type that all textures bottleneck through, such that the things currently encoded in the "flavor" bits can become explicit generic parameters/arguments.

Something like:
```
enum TextureShape { 2D, 3D, ... 2DArray, ... }
enum TextureAccess { ReadOnly, WriteOnly, ReadWrite, Append, Consume, ... }
enum Format { /* DXGI_FORMAT-style stuff... */ }
interface ITexelType { ... } // conformances for floatN, intN, etc.
interface ITextureFormat
{
    associatedtype Texel : ITexelType;
    associatedconst format : TextureFormat;
}

struct __Texture< let A : TextureAccess, let S : TextureShape, F : ITextureFormat >
{ ... }
```